engine/swarm/secrets.md

```diff
@@ -0,0 +1,771 @@
+    old version.
+
+    ```bash
+    $ mkdir tempdir && openssl rand -base64 20 > tempdir/mysql_password.txt
+
+    $ docker secret create -f tempdir/mysql_password.txt mysql_password
+
+    lwxgks2rojdbvaui5kay1suvr
+    ```
+
+    The value returned is not the password, but the ID (digest) of the secret.
+    List the secrets managed by Docker using docker secret ls:
+
+    ```bash
+    $ docker secret ls
+
+    ID                          NAME                CREATED             UPDATED             SIZE
+    lwxgks2rojdbvaui5kay1suvr   mysql_password      41 seconds ago      41 seconds ago      29 B
+    ```
+
+    The secret is stored in the encrypted WAL logs for the swarm.
+
+2.  Create the MySQL container.  The MySQL container will have the following
+    characteristics:
+
+    - Exposes the port publicly because otherwise the WordPress service will not
+      be able to access MySQL. Non-service containers created with `docker run`
+      cannot join swarm overlay networks by default.  
+    - Persists MySQL data in a named volume (called `tempdir`) so that data
+      isn’t lost across restarts.  
+    - The `tempdir` volume containing the `mysql_password.txt` secret will be
+      mounted as a volume in the container so that the password is never exposed
+      as an environment variable, nor can it be committed to an image if the
+      `docker commit` command is run.
+    - Sets the environment variable `MYSQL_ROOT_PASSWORD_FILE` to point to the
+      `mysql` password in the bind-mounted `tempdir` volume. The `mysql` image
+      reads the password string from that file when initializing the database
+      for the first time. Afterward, it is stored in the MySQL database itself.
+
+      ```bash
+      $ docker run --name mysql -d \
+                   -v $(pwd)/tempdir:/initpw \
+                   -e MYSQL_ROOT_PASSWORD_FILE=/initpw/mysql_password.txt \
+                   -p 3306:3306 \
+                   mysql:latest
+
+      67581e0d214168458f222f7de4d5566cf7de0c4022a4995f3efa323d31e16743
+      ```
+
+3.  Verify that the `mysql` container is running using the `docker ps` command.
+
+    ```bash
+    $ docker ps --filter name=mysql
+
+    CONTAINER ID        IMAGE               COMMAND                  CREATED         STATUS          PORTS                    NAMES
+    67581e0d2141        mysql:latest        "docker-entrypoint..."   8 minutes ago   Up 8 minutes    0.0.0.0:3306->3306/tcp   mysql
+    ```
+
+4.  Get the IP address of your host machine's main interface, likely `eth0` on
+    Linux hosts. You'll need it in the next step.
+
+5.  Now that MySQL is set up, create a WordPress service that connects to the
+    MySQL service. The WordPress service has the following characteristics:
+
+    - Because the scale is set to `1`, only a single WordPress task runs.
+      Load-balancing WordPress is left as an exercise to the reader, because of
+      limitations with storing WordPress session data on the container
```

Comment:
Should look into this again, maybe I was mistaken about this and it stores them in the database; let me know if you want me to find out